docker/README.md

```diff
@@ -1,93 +1,158 @@
+or Linux desktop or laptop.
 
-To build an image for a locally modified version of Trino Gateway, run the Maven
-build as normal for the `gateway-ha` modules, then build the image:
+```shell
+export TRINO_GATEWAY_IMAGE="trino-gateway:6-SNAPSHOT-amd64"
+```
 
-```bash
-./build.sh
+Use a published image from DockerHub.
+
+```shell
+export TRINO_GATEWAY_IMAGE="trinodb/trino-gateway:latest"
 ```
+The release process publishes images for Trino Gateway 6 and newer to DockerHub.
 
-The Docker build process will print the ID of the image, which will also
-be tagged with `trino-gateway:xxx-SNAPSHOT-yyy`, where `xxx-SNAPSHOT` is the version
-number of the Trino Maven build and `-yyy` is the platform the image was built for.
+Next set the image and platform:
 
-To build an image for a specific released version of Trino Gateway,
-specify the `-r` option, and the build script will download
-all the required artifacts:
+Start Trino Gateway and its PostgreSQL backend database, and wait until the 
+health check is successful:
 
 ```bash
-./build.sh -r 4
+docker compose up --wait
 ```
 
-Set the environment variable `TRINO_GATEWAY_BASE_IMAGE` to use a specific base image
-to build Trino Gateway image.
+Inspect the logs for progress and troubleshooting:
 
 ```bash
-export TRINO_GATEWAY_BASE_IMAGE=<image>
-./build.sh
+docker compose logs gateway
+```
+
+Typically your operating system automatically sets the default Docker platform
+value. Try to In some cases it can be useful to explicitly set it.
+
+For example on ARM64-based MacOS you can set it to use `linux` because it 
+otherwise is potentially set to `darwin` and there are no PostgreSQL images 
+available for `darwin` and this can prevent starting the with docker compose.
+
+```shell
+export DOCKER_DEFAULT_PLATFORM="linux/arm64"
 ```
 
-## Getting Help
+You can also set platform without operating system identifier:
+
+```shell 
+export DOCKER_DEFAULT_PLATFORM="amd64"
+```
+
+### Running
+
+Once everything is up and running, you can use the REST API to show the
+configured backends:
+
+```bash
+curl localhost:8080/api/public/backends
+```
+
+The Trino Gateway is available at [http://localhost:8080](http://localhost:8080).
+
+The PostgreSQL backend database for Trino Gateway runs at
+[http://localhost:5432](http://localhost:5432).
+
+## Configuration
+
+The image uses the configuration file `gateway-ha/gateway-ha-config-docker.yml`
+from the project checkout, and mounts it at `/opt/trino/gateway-ha-config.yml`.
+
+TBD add info or link to info about adding Trino clusters for local testing and in general
+
+## Health check
+
+By default the container health check uses the file `docker/bin/health-check`
+mounted at `/usr/lib/trino/bin/health-check`. The scripts expects a 2XX response
+from the server  at `/api/public/backends`.
+
+## Release
+
+After the container images are built locally, a maintainer can login to
+DockerHub and use the `release-docker.sh` script to publish the locally built
+images.
```

Comment:
This section is unrelated to most users. Not sure if worth mentioning.

Reply:
Because currently this is the readme file and it contains a mixture of user, developer and other docs. In a follow up we will split it apart and move into the proper docs